File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.py
from collections import deque

def solution(food):
    answer = ''
    front_deck = deque()
    back_deck = deque()
    
    for i in range(1, len(food)):
        cnt = food[i] // 2
        
        for _ in range(cnt):
            front_deck.append(i)
            back_deck.append(i)
    
    front = ''
    back = ''
    while front_deck:
        front += str(front_deck.popleft())
        back += str(back_deck.pop())
    
    answer = front + '0' + back
    
    return answer

File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.java
import java.util.*;

class Solution {
    public String solution(int[] food) {
        String answer = "";
        String front_str = "";
        String back_str = "";
        Deque<String> front = new ArrayDeque<>();
        Deque<String> back = new ArrayDeque<>();
        
        for(int i = 1; i < food.length; i++){
            int cnt = food[i] / 2;
            
            for(int j = 0; j < cnt; j++){
                String element = Integer.toString(i);
                front.addFirst(element);
                back.addFirst(element);
            }
        }
        
        while(!front.isEmpty()){
            front_str += front.pollLast();
            back_str += back.pollFirst();
        }
        
        answer = front_str + "0" + back_str;
        
        return answer;
    }
}